server/auth.go

```diff
@@ -152,6 +157,10 @@ func AuthorizedUser(
 		}
 
 		if hasAuthorizedRole(u, role) {
+			// use the first role, since there should only ever be one
+			// for any particular organization and hasAuthorizedRole
+			// should ensure that at least one role for the org exists
+			ctx = context.WithValue(ctx, roles.ContextKey, u.Roles[0].Name)
```

Comment:
per your comment here, are we guaranteed that in the context of an organization, the roles will have already been filtered, and thus the `0`th is the right one? seems like it would simply be safer to iterate over the roles and find the one that matches, which will basically be the same time as this direct accessor but will be safer in case this is ever not true

Reply:
> per your comment here, are we guaranteed that in the context of an organization, the roles will have already been filtered, and thus the 0th is the right one? 

there should only ever be one, but its probably still safer to iterate through them just in case.

Reply:
What do you think is the best path forward here?

Reply:
I'd rather not potentially obscure information here, so I'd prefer to check that the length is 0, and if so then use the 0th element, and if the len is not 0 then communicate at least via a log (or possibly an error) that somehow they've ended up with more than 1 role and to please report it on github, and still use the first one, what it is, and what it's using.